actionkit/api/mailer.py

```diff
@@ -0,0 +1,150 @@
+import json
+import re
+import requests
+import sys
+
+from actionkit.api.base import ActionKitAPI
+
+class AKMailerAPI(ActionKitAPI):
+
+    def update_mailing(self, mailing_id, update_dict):
+        res = self.client.patch(
+            #the '/' at the end is IMPORTANT!
+            '%s/rest/v1/mailer/%s/' % (self.base_url, mailing_id),
+            data=json.dumps(update_dict))
+        return self._http_return(res)
+
+    def get_mailing(self, mailing_id):
+        if getattr(self.settings, 'AK_TEST', False):
+            return TEST_DATA.get(mailing_id)
+        res = self.client.get(
+            #the '/' at the end is IMPORTANT!
+            '%s/rest/v1/mailer/%s/' % (self.base_url, mailing_id))
+        return {'res': res,
+                'mailing': res.json() if res.status_code == 200 else None}
+
+    def create_mailing(self, mailing_dict):
+        if getattr(self.settings, 'AK_TEST', False):
+            return TEST_DATA.get(mailing_dict.get('create_mailing'))
+        res = self.client.post(
+            '%s/rest/v1/mailer/' % self.base_url,
+            json=mailing_dict)
+        rv = {'res': res}
+        if res.headers.get('Location'):
+            rv['id'] = re.findall(r'(\d+)/$', res.headers['Location'])[0]
+        return rv
+
+    def get_mailings(self, **kwargs):
+        if getattr(self.settings, 'AK_TEST', False):
+            return TEST_DATA.get(1234)
+        params = []
+        for key, value in kwargs.items():
```

Comment:
I think we should do query string encoding with urlencode (https://stackoverflow.com/questions/7801991/how-to-convert-a-dictionary-to-query-string-in-python) to both make the code more readable and more correct (e.g. this doesn't seem to be URL-encoding).

Reply:
done